vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -1251,17 +1272,45 @@ void AdsImpl::GenerateAdReportingSustainEvent(
   writer.StartObject();
 
   writer.String("type");
-  writer.String("sustain");
+  writer.String("confirmation");
 
   writer.String("stamp");
-  std::string time_stamp = helper::Time::TimeStamp();
+  auto time_stamp = helper::Time::TimeStamp();
   writer.String(time_stamp.c_str());
 
   writer.String("notificationId");
   writer.String(info.uuid.c_str());
 
+  std::string type;
+  switch (info.type) {
+    case ConfirmationType::UNKNOWN: {
```

Comment:
can we extract all this types into variable? (same as comment above)

Reply:
👍 fixed